anesthetic/weighted_pandas.py

```diff
@@ -1,12 +1,131 @@
 """Pandas DataFrame and Series with weighted samples."""
 
+import warnings
 from inspect import signature
 import numpy as np
 from pandas import Series, DataFrame, concat, MultiIndex
+from pandas.core.groupby import GroupBy, SeriesGroupBy, DataFrameGroupBy, ops
+from pandas._libs import lib
+from pandas._libs.lib import no_default
+from pandas.util._exceptions import find_stack_level
 from pandas.util import hash_pandas_object
 from numpy.ma import masked_array
 from anesthetic.utils import (compress_weights, channel_capacity, quantile,
                               temporary_seed, adjust_docstrings)
+from pandas.core.dtypes.missing import notna
+
+
+class WeightedGroupBy(GroupBy):
+    """Weighted version of ``pandas.core.groupby.GroupBy``."""
+
+    grouper: ops.BaseGrouper
+    """:meta private:"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _add_weights(self, name, *args, **kwargs):
+        result = self.agg(lambda df: getattr(self.obj._constructor(df), name)
+                          (*args, **kwargs)).set_weights(self.get_weights())
+        return result.__finalize__(self.obj, method="groupby")
+
+    def mean(self, *args, **kwargs):  # noqa: D102
+        return self._add_weights("mean", *args, **kwargs)
+
+    def std(self, *args, **kwargs):  # noqa: D102
+        return self._add_weights("std", *args, **kwargs)
+
+    def median(self, *args, **kwargs):  # noqa: D102
+        return self._add_weights("median", *args, **kwargs)
+
+    def var(self, *args, **kwargs):  # noqa: D102
+        return self._add_weights("var", *args, **kwargs)
+
+    def kurt(self, *args, **kwargs):  # noqa: D102
+        return self._add_weights("kurt", *args, **kwargs)
+
+    def kurtosis(self, *args, **kwargs):  # noqa: D102
+        return self._add_weights("kurtosis", *args, **kwargs)
```

Comment:
Why do we need to define `kurt` and `kurtosis` here, but don't need to define `skew`? Yet `skew` still passes the tests...? 

I first thought this had to do with pandas `GroupBy` not overriding `skew` as opposed to `mean`, `median`, `std` and `var`. However, the same would go for `kurt` and `kurtosis`, so now I wonder what is different/special about `skew`?

Reply:
fyi I've found that `skew` needs adding for pandas 2 (otherwise it fails the is weighted test), so could add it in here now

Reply:
A little bit of an odd one -- pandas 1.5 doesn't actually implement groupby kurt and kurtosis (despite the fact I believe it could). @AdamOrmondroyd does it need adding for pandas 2? My instinct is to leave it in, as it doesn't do any harm.